pkg/reconciler/pipelinerun/resources/apply.go

```diff
@@ -318,9 +318,15 @@ func replaceParamValues(params []v1beta1.Param, stringReplacements map[string]st
 func ApplyTaskResultsToPipelineResults(
 	results []v1beta1.PipelineResult,
 	taskRunResults map[string][]v1beta1.TaskRunResult,
-	customTaskResults map[string][]v1beta1.CustomRunResult) ([]v1beta1.PipelineRunResult, error) {
+	customTaskResults map[string][]v1beta1.CustomRunResult,
+	skippedTasks []v1beta1.SkippedTask) ([]v1beta1.PipelineRunResult, error) {
 	var runResults []v1beta1.PipelineRunResult
 	var invalidPipelineResults []string
+	skippedTaskNames := map[string]bool{}
+	for _, t := range skippedTasks {
+		skippedTaskNames[t.Name] = true
+	}
+
```

Comment:
can we please add a log message or something to communicate to users that a pipelineresult references a skipped taskresult so the value cannot be substituted?

Reply:
sure

Reply:
added a log message when we skip the skipped task result reference

Reply:
The controller log file won't typically be accessible by users, and having a log entry in the "info" level about this feels like something that might spam logs if users often skipped tasks. I would rather have a debug log level here.

If we want to let users know about this, perhaps we could replace the result reference (or extend it) with a comment, saying that the result is not available as the task was skipped, but I'm not sure if that'd be a good idea.

Reply:
Oh sorry I missed your comment! @afrittoli 
I will change this to debug level log.